install.sh|upload.sh: Parse arguments without getopts | Improve input fle/folder parsing | Misc

Akianonymus opened the pull request:
e.g: gupload -d -o a b, here a is file/folder, b is gdrive folder.

   : gupload a b, here a is file/folder, b is gdrive folder.

   : gupload -d -o a b -s c d, here a, b and c are file/folder, d is gdrive folder.

* _json_value: Don't check for arguments

    because it works on pipe output too

* _full_path: Improve the function for various usescases, make it more accurate

* fix some typos

* Update README

labbots commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5571879&bid=18269338)

labbots commented on install.sh:
```diff
@@ -172,10 +170,13 @@ _dirname() {
+    [[ $# = 0 ]] && printf "%s: Missing arguments\n" "${FUNCNAME[0]}" && return 1
+    declare input="${1}"
+    if [[ -f ${input} ]]; then
+        printf "%s/%s\n" "$(cd $(_dirname "${input}") && pwd)" "${input##*/}"
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Quote this to prevent word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5571879)

labbots commented on install.sh:
```diff
@@ -172,10 +170,13 @@ _dirname() {
+    if [[ -f ${input} ]]; then
+        printf "%s/%s\n" "$(cd $(_dirname "${input}") && pwd)" "${input##*/}"
+    elif [[ -d ${input} ]]; then
+        printf "%s\n" "$(cd ${input} && pwd)"
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5571879)

labbots commented on utils.sh:
```diff
@@ -210,10 +210,13 @@ _extract_id() {
+    if [[ -f ${input} ]]; then
+        printf "%s/%s\n" "$(cd $(_dirname "${input}") && pwd)" "${input##*/}"
+    elif [[ -d ${input} ]]; then
+        printf "%s\n" "$(cd ${input} && pwd)"
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Double quote to prevent globbing and word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5571879)

labbots commented on utils.sh:
```diff
@@ -210,10 +210,13 @@ _extract_id() {
+    [[ $# = 0 ]] && printf "%s: Missing arguments\n" "${FUNCNAME[0]}" && return 1
+    declare input="${1}"
+    if [[ -f ${input} ]]; then
+        printf "%s/%s\n" "$(cd $(_dirname "${input}") && pwd)" "${input##*/}"
```
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Quote this to prevent word splitting.](https://app.codacy.com/manual/labbots/google-drive-upload/pullRequest?prid=5571879)

labbots commented:
@Akianonymus what is the need to change the use of getopts? if the user wants to upload multiple files then it could be handled by passing multiple -f option. This I believe is the standard for most linux commands. Handling it through unnamed argument doesn't seam to be the right method. So could you be please elaborate the reason for the change.

Akianonymus commented:
@labbots First, i removed getopts to decrease code lines.

Now about handling arguments, i was trying to match it to advance programs out there. e.g: curl handles multiple arguments like that, although it still have `--next` option for handling multiple arguments.

Note: While i implemented that, the previous behaviour is still there, i.e

gupload filename gdrive_folder someflags

I did this basically to we can handle arguments given between flags, as i have explained in the readme.

To sum up, yes this was basically an attempt to match argument handling like curl, and yes it is convenient to have multiple arguments without using -f option.

I believe there is nothing wrong in having multiple choices within disturbing the existing ones. Anyway, will remove if you say, i don't mind. :p

Akianonymus commented:
Anyway, do not merge now, need to push some fixes.
